oracleClient/src/main/kotlin/com/oracleClientStatesAndContracts/states/DiceRollState.kt

```diff
@@ -28,4 +18,49 @@ data class DiceRollState(
         val gameBoardStateUniqueIdentifier: UniqueIdentifier,
         override val participants: List<Party>,
         val signedDataWithOracleCert: SignedDataWithCert<Party>
-): ContractState, PersistentState()
\ No newline at end of file
+) : ContractState, PersistentState() {
+
+    constructor(diceRollState: DiceRollState) : this(
+            diceRollState.randomRoll1,
+            diceRollState.randomRoll2,
+            diceRollState.turnTrackerUniqueIdentifier,
+            diceRollState.gameBoardStateUniqueIdentifier,
+            diceRollState.participants,
+            diceRollState.signedDataWithOracleCert
+    )
+
+    init {
+        require(MIN_ROLL <= randomRoll1) {
+            "Roll has to be >= $MIN_ROLL"
+        }
+        require(randomRoll1 <= MAX_ROLL) {
+            "Roll has to <= $MAX_ROLL"
+        }
+    }
+
+    companion object {
+        const val MIN_ROLL = 1
+        const val MAX_ROLL = 6
+    }
+
+    fun getRollTrigger() = RollTrigger(randomRoll1 + randomRoll2)
+}
+
+@CordaSerializable
+data class RollTrigger(val real: Boolean, val total: Int) {
+
+    constructor() : this(false, 0)
+    constructor(total: Int) : this(true, total)
+
+    init {
+        require(real || total == 0) {
+            "Not real one should have a total of 0"
+        }
+        require(!real || 2 * DiceRollState.MIN_ROLL <= total) {
+            "A real one's total has to larger than ${DiceRollState.MIN_ROLL * 2}"
+        }
+        require(!real || total <= 2 * DiceRollState.MAX_ROLL) {
+            "A real one's total has to smaller than ${DiceRollState.MAX_ROLL * 2}"
+        }
```

Comment:
@rogersanick I created this `RollTrigger`, that has 2 situations. Wonder what is better. This or using a nullable or using an interface with 2 implementations.